debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -298,14 +299,18 @@
         String expectedTz = "2016-11-04T11:51:30Z"; //timestamp is stored with TZ, should be read back with UTC
         int expectedDate = Date.toEpochDay(LocalDate.parse("2016-11-04"), null);
         long expectedTi = LocalTime.parse("13:51:30").toNanoOfDay();
+        long expectedTiPrecision = LocalTime.parse("13:51:30.123").toNanoOfDay();
```

Comment:
I think this one reveals an existing bug. In the docs we say that `TIME` columns will be emitted as fields of type `io.debezium.time.Time` or `io.debezium.time.MicroTime` (depending on the precision mode and the precision of the column). But here we actually expect values in nano-seconds.

It's surprising though that this went unnoticed so far. The issue seems to be some confusion of scale and length, most probably in `JdbcValueConverters#converter()`. It needs care when fixing though, so to not mess things up for MySQL (wondering whether MySQL and PG deliver the two values in different order when querying for them in the metadata).

Reply:
@gunnarmorling I've checked the code - it is postgres specific and prescribes using nanoseconds when length is not defined or larger than 6. So maybe this is just deficiency in documentation.

Reply:
@jpechane The question is what we consider as the authoritative source of tuth, the existing code or the documented behavior? I'm in favor of the latter, i.e. TIME(1-3) should be represented by `Time` and TIME(4-6) by `MicroTime`. That's also consistent with the behavior for MySQL (documented and actual, AFAICS).

Reply:
I've filed [DBZ-709](https://issues.jboss.org/browse/DBZ-709) for that.